File: frontend/src/domain/product/Product.ts
export interface Product {
  id: string
  name: string,
  price: string,
  description: string,
  imageSrc: string,
  imageAlt: string,
  quantity: number
}

export class ProductModel implements Product {
  public id: string = '';

  constructor(public name: string,
              public price: string,
              public description: string,
              public imageSrc: string,
              public imageAlt: string,
              public quantity: number) {
  }
}

File: inventory-lib/src/main/java/mart/mono/inventory/lib/Product.java
package mart.mono.inventory.lib;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.UUID;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    private UUID id;

    private Catalog catalog;

    private String name;
    private String price;
    private String description;
    private String imageSrc;
    private String imageAlt;

    private int quantity;
}
